### PR TITLE
Fix QT5 to work as a backend for matplotlib

### PR DESCRIPTION
**Pull request by rchiodo**
Using ```%matplotlib <backend>``` is actually handled by IPython itself. What this does is changes the event loop in use to handle messages.

The QT event loop doesn't seem to queue messages so if we send more than one requestExecute at a time, it will hang.

Ideally we'd modify IJupyterSession to force all requests to go through a queue, but it doesn't return a promise. We'd have to implement our own custom future that enforced waiting for the previous future before making a request. 

This change seems to have fixed it for me and is much simpler. Going to try this first.

For #8322

### CODE REVIEW

File: src/client/datascience/notebook/vscodeNotebookController.ts [hunk 1]
```diff
@@ -325,7 +325,14 @@ export class VSCodeNotebookController implements Disposable {
             join(this.context.extensionPath, 'out', 'datascience-ui', 'ipywidgetsKernel', 'require.js'),
             join(this.context.extensionPath, 'out', 'ipywidgets', 'dist', 'ipywidgets.js'),
             join(this.context.extensionPath, 'out', 'datascience-ui', 'ipywidgetsKernel', 'ipywidgetsKernel.js'),
-            join(this.context.extensionPath, 'out', 'fontAwesome', 'fontAwesomeLoader.js')
+            join(this.context.extensionPath, 'out', 'fontAwesome', 'fontAwesomeLoader.js'),
+            join(
```
Comment (IanMatthewHuff) [inline src/client/datascience/notebook/vscodeNotebookController.ts:329]: What's this script addition here?

Comment (rchiodo) [reply src/client/datascience/notebook/vscodeNotebookController.ts:329]: Oh crap. Forgot about this. I was trying to debug something before. Thanks. I'll remove it.

File: src/client/datascience/ipywidgets/ipyWidgetMessageDispatcher.ts [hunk 2]
```diff
@@ -104,8 +104,8 @@ export class IPyWidgetMessageDispatcher implements IIPyWidgetMessageDispatcher {
     }
 
     public receiveMessage(message: IPyWidgetMessage): void {
-        if (process.env.VSC_JUPYTER_LOG_IPYWIDGETS && message.message.includes('IPyWidgets_')) {
```
Comment (rchiodo) [inline src/client/datascience/ipywidgets/ipyWidgetMessageDispatcher.ts:107]: I should remove this one too.

File: src/client/datascience/notebook/vscodeNotebookController.ts (hunk 1 above)
Comment (IanMatthewHuff) [reply src/client/datascience/notebook/vscodeNotebookController.ts:329]: Yeah, looked like a debug something. Rest looks good.

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**